### ISSUE TITLE
regression in pathbar

### ISSUE THREAD

**Issue by rcrath** [linuxmint/nemo#3147]
```
 * Nemo version (nemo --version) 
 ** 5.6.1

 * Is issue with desktop or windowed nemo? 
 ** windowed

 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) 
 ** manjaro

 * Graphics hardware *and* driver used: Intel CoffeeLake-S GT2 [UHD Graphics 630] vendor: Dell driver: i915 v: kernel,     arch: Gen-9.5 ports: active: DP-2,HDMI-A-1 empty: DP-1, DP-3, HDMI-A-2, HDMI-A-3 bus-ID: 00:02.0,     chip-ID: 8086:3e98
 
 * 32 or 64 bit
 ** 64bit


 ```
**Issue**
The main part of this is a regression passed off as a feature. The default behavior I and apparently others have used for years is to stay on breadcrumbs and only switch to the text path when I need to. I navigate folders that way all the time. There was a setting to keep this behavior that has been removed intentionally (see https://github.com/linuxmint/nemo/issues/3134#issuecomment-1345348258) because @mtwebster likes it another way, which thus far does not work as he says it does, is poorly documented, with rushed changes being pushed without checking them and issues closed while things are still broken.

I would like to remind @clefebvre and @mtwebster that the driving force behind creating cinnamon and nemo was changes made in in Gnome and in particular Nautilus that devs thought were great and made everyone else adhere to despite repeated requests to restore the working behavior that users needed.  Here it is in @clefebvre 's [own words](https://www.omgubuntu.co.uk/2012/09/linux-mint-explain-nautilus-fork-call-new-version-a-catastrophe). This is the same behavior repeated but with even less vetting, discussion, and use-testing than the changes that led to the forking of Nautilus 3.6 that made Nemo in the first place.

from the arch wiki: 

> [Nemo](https://github.com/linuxmint/nemo) is a fork of [GNOME Files](https://wiki.archlinux.org/title/GNOME_Files) (formerly known as Nautilus). It is also the default file manager of the [Cinnamon](https://wiki.archlinux.org/title/Cinnamon) desktop. Nemo is based on the Files 3.4 code. It was created as a response to the changes in Files 3.6 which saw features such as type ahead find and split pane view removed.

**Steps to reproduce**

use nemo 5.6. No setting to default to breadcrumbs. Removed because dev says you can't have it: "This was deliberate. Prior to this, you were permanently in one or the other mode." Specifically, this breaks the desired workflow. And: "More options isn't really an option here. The idea of this is to make the automatic behavior appealing to everyone." it is not a new option being asked for, it is a restoration of the option that was removed that caused the regression. And if it is supposed to please everyone, it does not achieve this.  

**Expected behaviour**

Restore the setting to default to breadcrumbs and turn text mode on and off using ctl-L or the button. States are sticky, don't change until I change them.  5.6.1 breaks the ability to navigate back and forth on the breadcrumbs trail. Every time I click on a breadcrumb, especially when returning to a deeper folder that is still on the breadcrumb trail, I get a text bar instead, and have to hit enter as per the instructions from @mtwebster, which does not return to breadcrumbs as advertised, then click the cursor inside of the text path and type ctl-L.  None of this used to happen. When I worked in breadcrumbs, it was breadcrumbs, and when I worked in text paths, it was text.  One keystroke (Ctl-L) or one click change to the other mode until I wanted the first mode back, as per https://github.com/linuxmint/mint21.1-beta/issues/73#issuecomment-1347189653.

Besides @mtwebster liking the regression better, is there any call to change _to_ the regression? Is there any complaint besides @mtwebster 's that the old behavior is broken? Is there any positive feedback from anyone regarding the changes you have pushed? The github issues, while few, have been universally opposed to the change. THere are references to we, but so far I have only seen @mlwebster weighing in and pushing all the changes solo.

**Other information**
I suppose the answer on one level is that if I don't like it I should fork it and make my own, but I actually do my work, which is not coding, on Cinnamon, and depend on it not being randomly broken. I am not a coder beyond html/css/regex/bash and thus I am utterly dependent on the devs to provide a stable workflow, which in all the years of cinnamon and nemo, has been the case. So, to be clear, thanks to @clefebvre and @mtwebster and everyone else who makes cinnamon such an excellent desktop environment. Heading over to make (hopefully not my last) financial contri to do what I can do to support you all.  It would be great if people coould weigh in here. If I am alone in opposing this regression, I'll suck it up and get used to a new behavior, but so far I see every reason fortreating this as a regression a la Nautilus 3.6. Others?

**Comment by mtwebster** [comment]
- 10/31 - Initial commit for change - https://github.com/linuxmint/nemo/commit/7ed6889706b24297cf8dfdfe01ada3d64583372a
- 11/17 - Some further refinement based on internal discussions - https://github.com/linuxmint/nemo/commit/0ac19c02e9225372291ea2a01370eb415b557956
- 11/18 - More refinement - https://github.com/linuxmint/nemo/commit/ae4262094804c396e22dd9a227b2297f7e27fe7a
- 11/21 - Feature freeze (5.6 tagged)
- 12/6 - Beta
- 12/12 - Some changes based on user feedback - https://github.com/linuxmint/nemo/commit/003e516cdaadd9a7d5ce0031c0280f2e51b779e0
- 12/20 - 21.1 Release.

Three weeks this stuff was merged before the feature freeze, and the beta lasted until around 12/15 or so. We got no feedback until beta, and when we did, we made changes. We put some effort this dev cycle into making it easier to test changes -
- We've begun to generate deb packages using github: https://github.com/linuxmint/nemo/releases. These are always built after every commit. It was touched on here: https://blog.linuxmint.com/?p=4405
- there's a PPA as well: https://launchpad.net/~linuxmint-daily-build-team/+archive/ubuntu/daily-builds.  It usually lags a day behind or so but it stays current.

I *WISH* I could get more people to try things out, and get more feedback. I will bend over backwards to help anyone that wants to try a new feature or fix out, and _always have_. But we're not going to have a public forum on every change or feature we work on. I say "we" a lot because there is usually some discussion behind the scenes, and, most importantly, there's no "I" in team.

> The default behavior I and apparently others have used for years is to stay on breadcrumbs and only switch to the text path when I need to.

This is how the default behavior works. Did you upgrade from 21? Perhaps the last mode you were in was entry. If so, check View->Toolbar and make sure 'Path Bar' is selected. That menu now lets you select the mode you want to be.. sticky? - the mode that gets returned to after a location change. That changes the actual 'setting' - which used to be changed by ctrl-L, etc.. in 5.4. Perhaps we should have created a new setting entry for that, so everyone would start in pathbar mode for the new version, or done something to make that slightly more visible. It seemed the best place for it  though, instead of being buried in the Preferences.

**Comment by rcrath** [comment]
Hi @mtwebster thanks for the detailed response, and I'm glad and reassured by what you wrote, that this is not a re-nautilus! 

TL;DR: Sorry, my bad this time. Arch release behind Mint release for first time ever.

It seems I have to wait a bit. Normally, Manjaro follows Arch quite regularly and mint releases generally arrive _after_ Arch updates. In this case, because the last fixes were so close to the mint 21 release date (12/20) mint got ahead of arch, which is still on 5.6.1-1 build of 12/13, which explains why I am still getting the broken behavior from the mistaken push. My apologies for just assuming and not checking the build date, which explains why I am not getting the documented behavior yet. I thought I was up-to-date because nemo doesn't seem to have an independent release number on github, just the mint version.  Anyway, I should have checked the build date instead of the version number before posting.  I flagged it out of date on Arch downstream, so hopefully it will float my way soon. 

So again, because I was vociferous in my complaint, I'll just as vociferously say sorry for jumping the gun and hope you will accept my apology. Also tagging @clefebvre  since I was clear to do so in the original.  Nautilus 3.6 PTSD. :)

**Comment by ArchangeGabriel** [comment]
@rcrath I can’t package a version that does not exist. 5.6.1 is the latest release (and master.mint21 is a moving tag…).

**Comment by rcrath** [comment]
@mtwebster I guess that means this is reopened. where can @ArchangeGabriel find a working version?. the 5.6.1 version which is the latest on arch is the mistaken push from Dec. 13 still. 5.6.2?

**Comment by mtwebster** [comment]
There is no 5.6.2 - 5.6.1 on December 13th is the most recent release.

I just tried nemo 5.6.1 on Manjaro (unfortunately that's the closest I can get to Arch currently) and it seemed to work fine to me. Did you check the setting in the View menu?

![image](https://user-images.githubusercontent.com/262776/210581011-dc6ef0b2-b4cc-431e-9cf4-099088e2ee74.png)

**Comment by DaveBlack** [comment]
Thank you @mtwebster for your time and development improvements,

Sorry unable to supply feedback during this last beta.

- Nemo 5.6.1
After testing the recent changes, wonder if a refinement might be... Once `Path Bar` has changed to `Location Entry`, clicking anywhere away from `Location Entry` inputbox automatically reverts back to `Path Bar`?

With the above modification, the 'Toggle Location Entry' menu option could possibly be hidden by default?

**Comment by mtwebster** [comment]
I've been considering that (reverting when focus is lost), though I also don't find it too hard to imagine that someone may want to view the path and maybe have another window focused temporarily (oops forgot to copy part of the path...) and then come back to it, so I'm not really sure yet.

**Comment by Jeremy7701** [comment]
I can't think of any other application that changes its appearance when a focus event occurs.
Definitely need to offer an option to disable this "smart" `Path Bar`.

**Comment by DaveBlack** [comment]
Thanks @mtwebster for your feedback,

Hmm, also considered that possibility.
One possible solution, revert when focus is lost, unless the 'Toggle Location Entry' toolbar icon has been deliberately used to  switch modes?

**Comment by rcrath** [comment]
Closing this issue, since 5.6.2 propagated to Arch and Manjaro and the original issue is solved.  If there are new issues with the pathbar plz refer back here and start a new issue if need be. Thank you @mtwebster for your hard work.
